tests/test_cap_api.py

```diff
@@ -526,6 +526,25 @@ def test_add_permissions(cap_api):
     assert resp == mock_permissions
 
 
+@responses.activate
+def test_add_permissions_no_access(cap_api):
+    upload_url = 'https://analysispreservation-dev.cern.ch/api/deposits/some-pid/actions/permissions'
+
+    responses.add(responses.POST, upload_url,
+                  content_type='application/json',
+                  headers={
+                      'Content-type': 'application/json',
+                      'Accept': 'application/permissions+json'
```

Comment:
Sorry, didn't notice before, but now can see you misused it in few tests already. You're confusing what `headers` param does. Those are the **response** headers, not the request ones. 

So to help to understand, in this example:
* you're sending content type json (when you're sending new permissions details), and accepting application/permissions+json (asking for updated serialized permission data for this analysis)
* server is gonna send you content type application/permissions+json in return (exactly what you asked for)

 Check some examples for responses library [here](https://github.com/getsentry/responses).

Reply:
if you need some more info for HTTP headers. just let me know;)